core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -146,18 +149,47 @@ public void onComplete() {
      */
     @Override
     public boolean tryComplete() {
-        topicPartitionDataFromTryComplete = acquirablePartitions();
+        if (anySharePartitionNoLongerExists()) {
```

Comment:
Hi @junrao, I am not actually too sure if I definitely need this check. Ideally, I don't think there can be any value in `sharePartitions` which can be null, but this [TODO](https://github.com/apache/kafka/blob/trunk/core/src/main/java/kafka/server/share/SharePartitionManager.java#L629) in `SharePartitionManager` is confusing me, plus there is this JIRA https://issues.apache.org/jira/browse/KAFKA-17510 where we will be refactoring share partition initialization. So, this can act as a safety check for now, and I can remove this in a future PR once the refactor is complete, and we are sure we don't send null share partition. What do you think?
cc - @apoorvmittal10

Reply:
yes, I'll remove `anySharePartitionNoLongerExists()` and also iterate on `sharePartitions` rather than using `shareFetchData.partitionMaxBytes()`. Will remove the null checks from our code.